aws/cloudformation.yml

```diff
@@ -0,0 +1,699 @@
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      VpcId: !Ref VPCStackName
+      GroupDescription: !Sub 'Redis SecurityGroup - ${UniqueID}'
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 6379
+          ToPort: 6379
+          CidrIp: !Sub '${EC2PrivateIP}/32'
+
+  RedisSubnetGroup:
+    Type: 'AWS::ElastiCache::SubnetGroup'
+    Properties:
+      Description: !Sub 'Redis SubnetGroup - ${UniqueID}'
+      SubnetIds:
+        - !Ref PublicSubnetA
+        - !Ref PublicSubnetB
+
+  RedisParameterGroup:
+    Type: 'AWS::ElastiCache::ParameterGroup'
+    Properties:
+      Description: !Sub 'Redis ParameterGroup - ${UniqueID}'
+      CacheParameterGroupFamily: redis4.0
+
+  ElastiCacheRedis:
+    Type: 'AWS::ElastiCache::CacheCluster'
+    Properties:
+      ClusterName: !Sub 'Mastodon-${UniqueID}'
+      CacheNodeType: cache.t2.micro
+      Engine: redis
+      EngineVersion: 4.0.10
+      Port: 6379
+      NumCacheNodes: '1'
+      CacheSubnetGroupName: !Ref RedisSubnetGroup
+      VpcSecurityGroupIds:
+        - !Ref RedisSecurityGroup
+      CacheParameterGroupName: !Ref RedisParameterGroup
+
+# RDS - PostgreSQL
+  RDSPostgreSQLSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      VpcId: !Ref VPCStackName
+      GroupDescription: !Sub 'PostgreSQL SecurityGroup - ${UniqueID}'
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 5432
+          ToPort: 5432
+          CidrIp: !Sub '${EC2PrivateIP}/32'
+
+  RDSPostgreSQLSubnetGroup:
+    Type: 'AWS::RDS::DBSubnetGroup'
+    Properties:
+      DBSubnetGroupDescription: !Sub 'PostgreSQL DBSubnetGroup - ${UniqueID}'
+      SubnetIds:
+        - !Ref PublicSubnetA
+        - !Ref PublicSubnetB
+
+  RDSPostgreSQLInstance:
+    Type: 'AWS::RDS::DBInstance'
+    DeletionPolicy: Snapshot
+    Properties:
+      Engine: postgres
+      Port: 5432
+      PubliclyAccessible: false
+      AllocatedStorage: 20
+      DBInstanceClass: db.t2.micro
+      DBInstanceIdentifier: !Sub 'Mastodon-${UniqueID}'
+      DBName: 'mastodon_production'
+      MasterUsername: !Ref RDSPostgreSQLMasterUsername
+      MasterUserPassword: !Ref RDSPostgreSQLMasterPassword
+
+      DBParameterGroupName: !Ref RDSPostgreSQLParameterGroup
+      DBSubnetGroupName: !Ref RDSPostgreSQLSubnetGroup
+      VPCSecurityGroups:
+        - !Ref RDSPostgreSQLSecurityGroup
+
+  RDSPostgreSQLParameterGroup:
+    Type: 'AWS::RDS::DBParameterGroup'
+    Properties:
+      Description: RDSPostgreSQL
+      Family: postgres9.6
+
+# S3 Storage Bucket - User/AccessKeys/Bucket
+  MasotodonS3bucket:
+    Type: 'AWS::S3::Bucket'
+    Properties:
+      BucketName: !Sub '${UniqueID}.files.${Domain}'
+
+  MasotodonS3buckerUser:
+    Type: 'AWS::IAM::User'
+    Properties:
+      Policies:
+        - PolicyName: S3
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action: 's3:*'
```

Comment:
We shouldn't need access to wildcard. I'd suggest limiting our scope to these:
- `ListBucket`
- `GetObject*`
- `PutObject*`
- `DeleteObject*`

Otherwise you're basically giving the IAM user the permission to run wild